src/Umbraco.Web/PublishedCache/NuCache/PublishedSnapshotService.cs

```diff
@@ -140,8 +142,11 @@ internal class PublishedSnapshotService : PublishedSnapshotServiceBase
                         lock (_storesLock)
                         {
                             _contentStore?.ReleaseLocalDb(); //null check because we could shut down before being assigned
+                            // Make sure the local dbs are disposed of if the content store was not created.
+                            _localContentDb?.Dispose();
```

Comment:
This is unecessary since this gets done as part of `_contentStore.ReleaseLocalDb`

Reply:
_contentStore has the Elvis check ...uuuhuh? so it may not have been created to release the local db so this is defensive to make sure of cleanup.

Reply:
Added further defences around the dispose calls.